test/io/nonblock/selector.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'fiber'
+
+class Selector
+  def initialize
+    @readable = {}
+    @writable = {}
+    @waiting = {}
+
+    @fiber = Fiber.current
+  end
+
+  attr :fiber
+
+  attr :readable
+  attr :writable
+  attr :waiting
+
+  def next_timeout
+    fiber, timeout = @waiting.sort_by{|key, value| value}.first
```

Comment:
Just a nit, but I think #min_by would be nice here:

```ruby
fiber, timeout = @waiting.min_by{|key, value| value}
```

Reply:
I will improve it.